### DISCUSSION TITLE
task splitting in case of Custom data extracts

### DISCUSSION THREAD

**Discussion by manjitapandey** [hotosm/field-tm#1405] [General]
How should we implement the task splitting algorithm when users upload custom data extracts?

One approach could be to allow users to upload both the data extracts and the linear features used to split tasks. In that case, We should inform users clearly that both should be included in a single file.

Another option can be to enable users to upload a custom data extract while extracting the linear features used for task splitting from OpenStreetMap (OSM).

**Comment by spwoodcock** [comment]
Good to get discussions started!

We already do option 2 and download linear features, but if none exist it doesn't help.

One thing we should do in fmtm is only enable splitting algo if data extract is polygons. Else, disable the button.

We also need to improve the algo error handling. If no linear features, it should just group as best it can by number of features. Thats an issue for the splitter repo

**Reply by Sujanadh** [reply]
Do we read the extracts to check if it contains linear feature or not? I don't think so as far i know. We extract data from osm if data_extracts is not passed to task-split consequently to split_by_sql.

**Reply by spwoodcock** [reply]
No we don't check if no linear features are returned.

This should ideally be handled in the splitting algorithm SQL.

If no linear features, then just group by the average number of buildings & return.

**Reply by spwoodcock** [reply]
@manjitapandey can we make an issue to only enable splitting algo if data extract contains polygons?

If no polygons are present, disable the splitting algorithm button, and possibly display a message below it to say 'the splitting algorithm is only valid for use with building data'.

**Comment by spwoodcock** [comment]
Discussion info taken from https://github.com/hotosm/fmtm/issues/1641

- Users should be able to upload a data extract of only buildings.
- The split by square and split by sql modes should 'just work', meaning we handle downloading the linestrings if necessary.

With this in mind, the data extract should only contain the features to be mapped (downloaded, or custom upload).

Then `split_by_square` should:
- Have the option in fmtm-splitter to take no data extract.
- However, we always pass through a data extract from FMTM, whether downloaded or custom upload.
- The data extract (containing only the features to be mapped) is used to determined if the square is removed from the final tasks (if no features are present in the square).

And `split_by_sql` should have user configuration on the frontend:
- Always take the data extract, either downloaded or custom upload.
- If the category is a type of linestring, e.g. mapping roads, we disable the splitting algorithm until we think of a solution.
- If the category is a type of polgon, e.g. buildings, we enable the splitting algorithm.
- We ask the user for input with a set of checkboxes for type of linestring they want to split by:
  - [ ] major roads
  - [ ] minor roads
  - [ ] rivers
  - [ ] railways
  - [ ] airstrips
- These linestrings are downloaded by `fmtm-splitter` during the splitting process, and the split tasks are returned.
